backend/src/api/bitcoin/bitcoin-api.ts

```diff
@@ -174,35 +191,20 @@ class BitcoinApi implements AbstractBitcoinApi {
       };
     }
 
-    if (transaction.confirmations) {
-      esploraTransaction = await this.$calculateFeeFromInputs(esploraTransaction, addPrevout);
-    } else {
-      esploraTransaction = await this.$appendMempoolFeeData(esploraTransaction);
-      if (addPrevout) {
-        esploraTransaction = await this.$calculateFeeFromInputs(esploraTransaction, addPrevout);
+    if (addPrevout) {
+      if (transaction.confirmations) {
+        esploraTransaction = await this.$calculateFeeFromInputs(esploraTransaction);
+      } else {
+        esploraTransaction = await this.$appendMempoolFeeData(esploraTransaction);
+        if (addPrevout) {
```

Comment:
This second `if (addPrevout)` is unnecessary because it's already nested in the exact same test